item_init/automation/jsr223/python/community/item_init/300_item_init.py

```diff
@@ -53,22 +53,30 @@ def item_init(event):
 
     item_init.log.info("Initializing Items")
     for item_name in [i for i in items if get_metadata(i, "init")]:
-        value = get_value(item_name, "init")
+        try:
+            value = get_value(item_name, "init")           
 
-        # Always update if override is True
-        if get_key_value(item_name, "init", "override") == "True":
-            post_update_if_different(item_name, value)
-            item_init.log.info("Overriding current value {} of {} to {}"
-                                .format(items[item_name], item_name, value))
+            # Always update if override is True
+            if get_key_value(item_name, "init", "override") == "True":
+                post_update_if_different(item_name, value)
+                item_init.log.info("Overriding current value {} of {} to {}"
+                                    .format(items[item_name], item_name, value))
 
-        # If not overridden, only update if the Item is currently NULL or UNDEF.
-        elif isinstance(items[item_name], UnDefType):
-            item_init.log.info("Initializing {} to {}"
-                                .format(item_name, value))
-            postUpdate(item_name, value)
+            # If not overridden, only update if the Item is currently NULL or UNDEF.
+            elif isinstance(items[item_name], UnDefType):
+                item_init.log.info("Initializing {} to {}"
+                                    .format(item_name, value))
+                postUpdate(item_name, value)
 
-        # Delete the metadata now that the Item is initialized.
-        if get_key_value(item_name, "init", "clear") == "true":
-            item_init.log.info("Removing init metadata from {}"
-                                .format(item_name))
-            remove_metadata(item_name, "init")
+            # Delete the metadata now that the Item is initialized.
+            if get_key_value(item_name, "init", "clear") == "true":
+                item_init.log.info("Removing init metadata from {}"
+                                    .format(item_name))
+                remove_metadata(item_name, "init")
+                
+        except:
```

Comment:
Which line or lines in the above raises the exception? I'd rather check for that condition and generate an actionable log statement, or use several smaller try/excepts than using a try/except with a generic "Sorry, couldn't do it." log message. That way it becomes more clear to the users how to correct their metadata to solve the problem.

Reply:
I have not been able to reproduce so far. I have tested the following:
on/off init tag for nb item type --> warning but rule completes
nb init tag for on/off  item type --> warning but rule completes
empty string init tag for number type item --> warning but rule completes

I will run without the try/except block for a few days to see if I can catch the exception again.

Reply:
That actually was the source of my question. If the value is not suitable for the Item, the events.postUpdate() call should generate a warning. I can't see anything here that actually generates an exception so I look forward to seeing what exception you saw. There is clearly something subtle or something here throws exceptions that I'm not aware of. 

From what I'm aware of:

- get_value return None if the metadata namespace of value doesn't exist
- get_key_value also returns None if the namespace or the key doesn't exist
- postUpdate I think doesn't raise an exception that we could catch
- we check to make sure the metadata exists before deleting it so that shouldn't produce an exception, though there might be room for error here as the check may be better to use get_metadata instead of assuming the value is defined. 

I'm actually OK leaving just the warning from postUpdate rather than trying to go through the pain of checking if the string is suitable for the Item type.  But if there is an exception that's possible, we should indeed catch it and explain how to fix it. 

Thanks for the PR!

Reply:
I agree with your comments, The warnings are explicit enough to guide the user in the right direction.
I have run for a few days and have not encountered the exception again.

I think you can close this PR for the moment. If I get the exception again I will try to do a better job in pinpointing.

Sorry if I am misusing the system I am new to this!
I have a couple more suggestions / improvements to submit.

Thanks for the great job!

Reply:
You are not misusing the system at all. Though this does show how combining multiple changes in one PR can become problematic. But it's not big deal. This isn't a huge project like openHAB. I can afford to be flexible. I'd rather have the PRs than people not submitting anything because they are afraid or don't know the process.

What I recommend doing is back out your changes to the code but keep the rename of the file. Then add a sentence or two to the README.md file explaining the 300 prefix and we can have this PR just include that change.

Reply:
I have submitted a commit to revert the try / except block and a second one for the README.md.

Let me know if you need an more changes.

Thanks for your patience